tests/integration/bank/keeper/deterministic_test.go

```diff
@@ -121,36 +118,25 @@ func (suite *DeterministicTestSuite) TestGRPCQueryBalance() {
 		suite.runQueryBalanceIterations(addr, &coin)
 	})
 
-	addr, err := sdk.AccAddressFromBech32("cosmos139f7kncmglres2nf3h4hc4tade85ekfr8sulz5")
-	suite.Require().NoError(err)
-
-	coin := sdk.NewCoin(
-		"denom",
-		sdk.NewInt(10),
-	)
-
-	suite.fundAccount(addr, coin)
-	suite.runQueryBalanceIterations(addr, &coin)
+	suite.fundAccount(addr1, coin1)
+	suite.runQueryBalanceIterations(addr1, &coin1)
 }
 
-func (suite *DeterministicTestSuite) runAllBalancesIterations(addr sdk.AccAddress, prevRes sdk.Coins) {
-	for i := 0; i < 1000; i++ {
-		res, err := suite.queryClient.AllBalances(suite.ctx, &banktypes.QueryAllBalancesRequest{
-			Address: addr.String(),
-		})
+func (suite *DeterministicTestSuite) runAllBalancesIterations(req *banktypes.QueryAllBalancesRequest, prevRes *banktypes.QueryAllBalancesResponse) {
+	for i := 0; i < iterCount; i++ {
+		res, err := suite.queryClient.AllBalances(suite.ctx, req)
 
 		suite.Require().NoError(err)
 		suite.Require().NotNil(res)
-		suite.Require().NotNil(res.Balances)
 
-		suite.Require().Equal(res.GetBalances(), prevRes)
-		prevRes = res.GetBalances()
+		suite.Require().Equal(res, prevRes)
 	}
 }
 
 func (suite *DeterministicTestSuite) TestGRPCQueryAllBalances() {
+	suite.SetupTest() // reset
```

Comment:
if we keep this out of the `rapid.Check` then previously added denoms won't be removed. always tests the pagination of first added denoms range(1,10)

Reply:
I'm not sure I understand. In each rapid test, we are creating one random `addr`, and adding 10 denoms for that `addr` only.

So each rapid test is independent in a sense, since the `addr` will be different.